src/BaGet.Core/Search/ISearchService.cs

```diff
@@ -33,16 +33,10 @@ public interface ISearchService
         /// Enumerate listed package versions.
         /// See: https://docs.microsoft.com/en-us/nuget/api/search-autocomplete-service-resource#enumerate-package-versions
         /// </summary>
-        /// <param name="packageId">The package ID.</param>
-        /// <param name="includePrerelease">Whether pre-release packages should be returned.</param>
-        /// <param name="includeSemVer2">Whether packages that require SemVer 2.0.0 compatibility should be returned.</param>
+        /// <param name="request">The autocomplete request.</param>
         /// <param name="cancellationToken">A token to cancel the task.</param>
         /// <returns>The package versions that matched the request.</returns>
-        Task<AutocompleteResponse> ListPackageVersionsAssync(
-            string packageId,
-            bool includePrerelease,
-            bool includeSemVer2,
-            CancellationToken cancellationToken);
+        Task<AutocompleteResponse> ListPackageVersionsAsync(AutocompleteRequest request, CancellationToken cancellationToken);
```

Comment:
The list package versions endpoint will always list *all* versions and doesn't accept `skip` or `take` parameters so I don't think we should reuse the `AutocompleteRequest` type here.

For more information, see these docs: https://docs.microsoft.com/en-us/nuget/api/search-autocomplete-service-resource#enumerate-package-versions

Reply:
I've made a separate VersionsRequest object, I know that there is quite a lot of code duplication inherent in doing that but it felt like it made it the easiest to understand